### PR TITLE
chore: use eslint rather than xo

### PR DESCRIPTION
**Pull request by Xotic750**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "main": "dist/lib/rule-finder.js",
   "scripts": {
     "cover": "nyc --reporter=lcov --reporter=text npm test",
-    "lint": "xo",
+    "lint": "eslint --ext .js,.json . --fix",
```
Comment (ljharb) [inline package.json]: typically autofix is applied explicitly, not implicitly; we wouldn’t want CI to pass when there’s fixable warnings in the code.

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
```
Comment (ljharb) [inline .eslintrc.js]: i prefer an `.eslintrc` file. that it’s “deprecated” officially is irrelevant; they’ll never remove it.

File: package.json [hunk 3]
```diff
@@ -83,12 +83,5 @@
   "bugs": {
     "url": "https://github.com/sarbbottam/eslint-find-rules/issues"
   },
-  "homepage": "https://github.com/sarbbottam/eslint-find-rules#readme",
-  "xo": {
-    "envs": [
-      "mocha"
```
Comment (ljharb) [inline package.json]: we’ll need to set the mocha env for test files as well.

Comment (Xotic750) [reply package.json]: `mocha` is set in the `.eslintrc`, or were you referring to something else?

File: .eslintrc [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
+  "env": {
+    "commonjs": true,
+    "es6": true,
+    "node": true,
+    "mocha": true
```
Comment (ljharb) [inline .eslintrc]: it’d be ideal to only enable this for test files, not everything else.

File: package.json [hunk 5]
```diff
@@ -5,7 +5,8 @@
   "main": "dist/lib/rule-finder.js",
   "scripts": {
     "cover": "nyc --reporter=lcov --reporter=text npm test",
-    "lint": "xo",
+    "lint": "eslint --ext .js,.json .",
+    "lint-fix": "npm run lint -- --fix",
```
Comment (ljharb) [inline package.json]: I don’t think this extra script is needed

File: .eslintrc (hunk 4 above)
Comment (Xotic750) [reply .eslintrc]: ok

File: package.json (hunk 3 above)
Comment (ljharb) [reply package.json]: it’s set for every file; i mean it should only apply to tests.

**Pull request merged**